relay-server/src/processing/logs/store.rs

```diff
@@ -75,16 +82,143 @@ fn ts(dt: DateTime<Utc>) -> Timestamp {
     }
 }
 
-fn attributes(attributes: Attributes) -> HashMap<String, AnyValue> {
-    let mut result = HashMap::with_capacity(attributes.0.len());
+/// The schema of a 'metadata attribute' stored in EAP.
+///
+/// A metadata attribute is a regular attribute stored in EAP, but it carries metadata about fields
+/// of the original payload processed by Relay and other components.
```

Comment:
Maybe a quibble, but "carries metadata" makes it sound to me like the metadata exists _in addition_ to the attribute value, whereas what is meant here is simply that the value of the attribute is the metadata of some field serialized to JSON.
Perhaps
```suggestion
/// A metadata attribute is a regular attribute stored in EAP that contains metadata about a field
/// of the original payload processed by Relay and other components.
```
?